ios/RNTweetShadowView.m

```diff
@@ -4,23 +4,24 @@ @implementation RNTweetShadowView
 
 - (TWTRTweetView *)tweetViewForMeasuring
 {
-    static TWTRTweetView *tweetView = nil;;
+    static TWTRTweetView *tweetView = nil;
     static dispatch_once_t onceToken;
     dispatch_once(&onceToken, ^{
         tweetView = [[TWTRTweetView alloc] initWithTweet:nil style:TWTRTweetViewStyleCompact];
     });
     return tweetView;
 }
 
-
 - (CGSize)computeIntrinsicSize:(CGFloat)width
 {
     static CGSize defaultSize;
     static dispatch_once_t onceToken;
     dispatch_once(&onceToken, ^{
-        TWTRTweetView *view = [self tweetViewForMeasuring];
-        defaultSize = [view sizeThatFits:CGSizeMake(width, CGFLOAT_MAX)]; // 200 is the minimum width required for a tweet
-        defaultSize.width = UIViewNoIntrinsicMetric;
+        dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
You are calling `dispatch_async` and in the block you are modifying `deafultSize`. `defaultSize` then gets immediately returned, i.e.you don't wait for your async call to finish.

Perhaps `dispatch_sync` is in order here?

Reply:
Of course, you’re right.